docs/import/manage_disaster.js

```diff
@@ -369,71 +370,67 @@ function addDisaster() {
  * Writes the given details to a new disaster entry in firestore. Fails if
  * there is an existing disaster with the same details.
  *
- * TODO(janakr): If the user starts editing a disaster before the Firestore
- *  write completes, their edit could be overwritten by the initial Firestore
- *  write here. Probably solved similar to the delete disaster issue: don't
- *  actually show the disaster as editable until this write completes.
  * @param {string} disasterId of the form <year>-<name>
  * @param {Array<string>} states array of state (abbreviations)
- * @return {Promise<boolean>} returns true after successful write to firestore
- * and earth engine folder creations.
+ * @return {Promise<boolean>} Returns true if EarthEngine folders created
+ *     successfully and Firestore write was successful
  */
-function writeNewDisaster(disasterId, states) {
+async function writeNewDisaster(disasterId, states) {
   if (disasterData.has(disasterId)) {
     setStatus('Error: disaster with that name and year already exists.');
-    return Promise.resolve(false);
+    return false;
   }
   clearStatus();
   const currentData = createDisasterData(states);
   disasterData.set(disasterId, currentData);
   // We know there are no assets in folder yet.
   disasterAssets.set(disasterId, Promise.resolve(new Map()));
 
-  const folderCreationPromises = [];
-  folderCreationPromises.push(
-      getCreateFolderPromise(eeLegacyPathPrefix + disasterId));
-  for (const state of states) {
-    folderCreationPromises.push(
-        getCreateFolderPromise(legacyStateDir + '/' + state));
+  const eeFolderPromises =
+      [getCreateFolderPromise(eeLegacyPathPrefix + disasterId)];
+  states.forEach(
+      (state) => eeFolderPromises.push(
+          getCreateFolderPromise(legacyStateDir + '/' + state)));
+
+  const tailError = '" You can try refreshing the page';
+  // Unsafe to do the Firestore write before EE folders are created, since maybe
```

Comment:
super nit: I get this phrasing knowing the history of this method but maybe just say the affirmative i.e. "wait on ee folder creation to do firestore write because..."

Reply:
Rephrased, how's this?